tests/analytics/test_forecasting_v2_episode_lifecycle.py

```
@@ -2214,3 +2269,1323 @@ def test_contradictory_creation_geometry_fails_closed():
+        operational_facts=(
+            good.operational_facts if operational_facts is _UNSET else operational_facts),
+        planned_risk=planned_risk)
+
+
+def _forge_from(good, *, outcome=_UNSET, new_state=_UNSET, reason=_UNSET, signal=_UNSET,
+                operational_facts=_UNSET, planned_risk=_UNSET) -> V2LifecycleDecision:
+    """General-purpose forge: clone every field of a REAL decision except
+    the ones explicitly overridden. Unlike `_forge_confirmed()`, the
+    outcome/new_state need not be CONFIRMED -- used for the
+    PRECONFIRMATION_UPDATE/INVALIDATED/EXPIRED forgery vectors."""
+    def pick(over, field):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `pick`

(ANN202)